src/control/cmd/daos_agent/infocache.go

```diff
@@ -292,6 +299,41 @@ func (c *InfoCache) EnableStaticFabricCache(ctx context.Context, nf *NUMAFabric)
 	c.EnableFabricCache()
 }
 
+func (c *InfoCache) getAttachInfo(ctx context.Context, rpcClient control.UnaryInvoker, req *control.GetAttachInfoReq) (*control.GetAttachInfoResp, error) {
+	if c == nil {
+		return nil, errors.New("InfoCache is nil")
+	}
+	if c.getAttachInfoCb == nil {
+		return nil, errors.New("getAttachInfoFn is nil")
+	}
+
+	resp, err := c.getAttachInfoCb(ctx, rpcClient, req)
+	if err != nil {
+		return nil, err
+	}
+	c.adjustAttachInfo(resp)
+	return resp, nil
+}
+
+// adjustAttachInfo performs any necessary adjustments to the attach info
+// before returning it.
+func (c *InfoCache) adjustAttachInfo(resp *control.GetAttachInfoResp) {
```

Comment:
We make other modifications to the GetAttachInfoResp struct in mgmt_rpc.go before returning to the client. Infocache has been caching only what the server sends back. I'm not exactly opposed to doing it this way, but it feels a little odd making our client-side modifications in two different layers.